### PR TITLE
rework getSearchValue() to take optional html param

### PR DESCRIPTION
**Pull request by andrewandante**
Fixes a problem where a field that returns a string (say, `getElementsForSearch()`) is treated as a `DBText` and when it has `->getSearchValue()` called on it, `->forTemplate()` calls `nl2br()` on the value and inserts a whole pile of HTML into our lovely clean field.

Now will strip it all out if the `include_page_html` config var is set to `false`

### CODE REVIEW

**Comment by madmatt** [comment]
Thanks @andrewandante! I'm torn on this. It doesn't _feel_ clean, but I struggle to come up with a better way of doing the same thing. @unclecheese Any chance you could take a look and see what you think about this please 🙏🏼?

**Comment by unclecheese** [comment]
Interesting. I can definitely see how this would be an annoying issue.

I'm thinking that the whole `shouldIncludeHTML` use case on DBFields is a bit dubious to begin with. Who really wants to search HTML tags? I can only see it being useful in one situation, which is if you want to show a preview of the search result. This option was really more intended for the page crawler service, and even then, I wonder about its value.

In any case, it seems appropriate that only `HTMLText` fields should qualify for this option, if we're going to apply it to `getSearchValue()`.

I would say:
* `DBField::getSearchValue()` -> sanitise
* `DBHTMLText::getSearchValue()` -> Check index configuration to determine whether to sanitise or not

It seems like HTML in your search content is a major edge case, and it would only be of value in the page crawler field(s). Is there a use case I'm missing?

**Comment by andrewandante** [comment]
I think I agree - I was thinking about this last night and trying to think of a scenario where HTML in your search is a good idea - the preview makes sense, but you're right, that only makes sense on a page-crawler level.

Will amend this PR to _always_ sanitise TextFields, and optionally sanitise HTMLFields. And throw some extra tests in.

Thanks!

**Comment by andrewandante** [comment]
I'd also argue that the config var should probably default to `false` - given that the inclusion of HTML is an edge-case, as you say

**Comment by andrewandante** [comment]
Alright I feel better about this now. Feels like it's doing what it should be.

File: src/Extensions/DBHTMLFieldExtension.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+     *
+     * @param bool $shouldIncludeHTML
+     * @return string|string[]|null
+     */
+    public function getSearchValue(bool $shouldIncludeHTML = true)
+    {
+        if ($shouldIncludeHTML) {
+            return $this->owner->forTemplate();
+        }
+        return preg_replace('/\s+/S', " ", strip_tags($this->owner->forTemplate()));
```
Comment (unclecheese) [inline src/Extensions/DBHTMLFieldExtension.php:22]: Let's just defer to the configuration here, rather than adding a new param. The param is only used in this one case, so I don't see a need for it (it's ignored in the generic implementation).

**Review by unclecheese — CHANGES_REQUESTED** [review]

File: src/Extensions/DBHTMLFieldExtension.php [hunk 2]
```diff
@@ -4,18 +4,28 @@
 namespace SilverStripe\SearchService\Extensions;
 
 use SilverStripe\Core\Extension;
+use SilverStripe\SearchService\Service\IndexConfiguration;
+use SilverStripe\SearchService\Service\Traits\ConfigurationAware;
 
 class DBHTMLFieldExtension extends Extension
 {
+    use ConfigurationAware;
```
Comment (unclecheese) [inline src/Extensions/DBHTMLFieldExtension.php]: If you want to do it with injection, I think you'll need another trait here. Extensions aren't injectable by default.

```suggestion
    use ConfigurationAware;
    use Injectable;
```

File: src/Extensions/DBHTMLFieldExtension.php [hunk 3]
```diff
@@ -4,18 +4,28 @@
 namespace SilverStripe\SearchService\Extensions;
 
 use SilverStripe\Core\Extension;
+use SilverStripe\SearchService\Service\IndexConfiguration;
+use SilverStripe\SearchService\Service\Traits\ConfigurationAware;
```
Comment (unclecheese) [inline src/Extensions/DBHTMLFieldExtension.php:9]:
```suggestion
use SilverStripe\Core\Injector\Injectable;
```

**Review by unclecheese — CHANGES_REQUESTED** [review]

File: src/Extensions/DBHTMLFieldExtension.php (hunk 2 above)
Comment (andrewandante) [reply src/Extensions/DBHTMLFieldExtension.php]: Oh I saw this after I pushed a different approach - is this more idiomatic? Seems not to work with the tests, so I swapped it for querying the `SearchServiceExtension` directly

**Comment by unclecheese** [comment]
Yup, I'd prefer the singleton you've used here. Nice job.

**Review by unclecheese — APPROVED** [review]
